### ISSUE TITLE
"TLSRequired off" plus Protocols directive causes mod_tls to terminate the session abruptly

### ISSUE THREAD

**Issue by viktor077** [proftpd/proftpd#1679]
### What I Did

My proftpd runs in a docker container based on Alpine linux.

### What I Expected/Wanted

I expect that user **ftpuser** to connect and transfer data without TLS/SSL and other users with TLS/SSL.

### ProFTPD Version and Configuration
```Compile-time Settings:
  Version: 1.3.9rc1 (git)
  Platform: LINUX [Linux 5.15.0-33-generic x86_64]
  OS/Release:
    NAME="Alpine Linux"
    ID=alpine
    VERSION_ID=3.18.0
    PRETTY_NAME="Alpine Linux v3.18"
  Built: Wed May 31 2023 09:45:01 UTC
  Built With:
    configure  '--build=x86_64-alpine-linux-musl' '--host=x86_64-alpine-linux-musl' '--prefix=/usr' '--sysconfdir=/etc/proftpd' '--localstatedir=/run/proftpd' '--disable-static' '--enable-sendfile' '--enable-facl' '--enable-dso' '--enable-autoshadow' '--enable-ctrls' '--disable-strip' '--disable-wtmp' '--disable-auth-pam' '--enable-openssl' '--with-shared=mod_ratio:mod_quotatab:mod_quotatab_sql:mod_readme:mod_sql:mod_sql_mysql:mod_sql_passwd:mod_tls:mod_ifsession' 'build_alias=x86_64-alpine-linux-musl' 'host_alias=x86_64-alpine-linux-musl'

  CFLAGS: -g2 -O2 -Wall -fno-omit-frame-pointer -fno-strict-aliasing -Werror=implicit-function-declaration
  LDFLAGS: -Wl,-L$(top_srcdir)/lib,-L$(top_builddir)/lib  -rdynamic  -L/usr/lib/
  LIBS:  -lssl -lcrypto

  Files:
    Configuration File:
      /etc/proftpd/proftpd.conf
    Pid File:
      /run/proftpd/proftpd.pid
    Scoreboard File:
      /run/proftpd/proftpd.scoreboard
    Header Directory:
      /usr/include/proftpd
    Shared Module Directory:
      /usr/libexec

  Info:
    + Max supported UID: 4294967295
    + Max supported GID: 4294967295

  Features:
    + Autoshadow support
    + Controls support
    - curses support
    - Developer support
    + DSO support
    + IPv6 support
    + Largefile support
    - Lastlog support
    - Memcache support
    - ncurses support
    - NLS support
    + OpenSSL support (OpenSSL 3.1.1 30 May 2023)
    - PCRE support
    - PCRE2 support
    + POSIX ACL support
    - Redis support
    + Sendfile support
    + Shadow file support
    - Sodium support
    + Trace support
    + xattr support

  Tunable Options:
    PR_TUNABLE_BUFFER_SIZE = 1024
    PR_TUNABLE_DEFAULT_RCVBUFSZ = 8192
    PR_TUNABLE_DEFAULT_SNDBUFSZ = 8192
    PR_TUNABLE_ENV_MAX = 2048
    PR_TUNABLE_GLOBBING_MAX_MATCHES = 100000
    PR_TUNABLE_GLOBBING_MAX_RECURSION = 8
    PR_TUNABLE_HASH_TABLE_SIZE = 40
    PR_TUNABLE_LOGIN_MAX = 256
    PR_TUNABLE_NEW_POOL_SIZE = 512
    PR_TUNABLE_PATH_MAX = 4096
    PR_TUNABLE_SCOREBOARD_BUFFER_SIZE = 80
    PR_TUNABLE_SCOREBOARD_SCRUB_TIMER = 30
    PR_TUNABLE_SELECT_TIMEOUT = 30
    PR_TUNABLE_TIMEOUTIDENT = 10
    PR_TUNABLE_TIMEOUTIDLE = 600
    PR_TUNABLE_TIMEOUTLINGER = 10
    PR_TUNABLE_TIMEOUTLOGIN = 300
    PR_TUNABLE_TIMEOUTNOXFER = 300
    PR_TUNABLE_TIMEOUTSTALLED = 3600
    PR_TUNABLE_XFER_SCOREBOARD_UPDATES = 10
```
**Full configuration in one file**
```
ServerName                  "ftp"
ServerType                  standalone
DefaultServer               on
ServerIdent                 on "ftp"             
DenyFilter                  \*.*/
AllowStoreRestart           on     
AllowRetrieveRestart        on  
ScoreboardScrub off
Port                        21
Umask                       002 002
MaxInstances                30
CommandBufferSize           512
User                        proftpd
Group                       proftpd
AllowOverwrite              on
AuthOrder mod_sql.c
Protocols ftp ftps
RequireValidShell   off
<Limit SITE_CHMOD FEAT>
    DenyAll
</Limit>
SocketOptions               keepalive on
MaxClients                  50 "Sorry, the maximum number of allowed users are already connected (%m)"
MaxClientsPerHost           10  "Sorry, you may not connect more than one time."
MaxLoginAttempts            3
UseReverseDNS               off
TimeoutIdle                 300
TimeoutLogin                300
TimeoutNoTransfer           360
TimeoutStalled              640
DefaultTransferMode         binary
AllowForeignAddress         off
UseReverseDNS               off
RootLogin                   off
AccessDenyMsg               "ATTENTION!!! ALL CONNECTIONS LOGED"
AccessGrantMsg              "Now upload/download files"
WtmpLog                                     off
LoadModule              mod_ratio.c
LoadModule              mod_quotatab.c
LoadModule              mod_quotatab_sql.c
LoadModule              mod_readme.c
LoadModule              mod_sql.c
LoadModule              mod_sql_mysql.c
LoadModule              mod_sql_passwd.c
LoadModule              mod_tls.c
LoadModule              mod_ifsession.c
SQLAuthTypes                SHA256
SQLPasswordEngine           on
SQLAuthenticate             users
SQLConnectInfo              %{env:FTP_DB_NAME}@%{env:FTP_DB_HOST} %{env:FTP_DB_USER} %{env:FTP_DB_PASS}
SQLUserInfo                 users username password uid gid homedir shell
SQLLogFile                  /var/log/proftpd/proftpd-sql.log
RequireValidShell           off
QuotaEngine                 on
QuotaDirectoryTally         on
QuotaDisplayUnits           Mb
QuotaShowQuotas             on
<IfModule mod_tls.c>
    TLSEngine                   on
    TLSOptions                  AllowPerUser
    TLSRequired                 auth+data
    TLSVerifyClient             off
    TLSProtocol                 ALL
    TLSOptions                  NoSessionReuseRequired
    TLSLog                      /var/log/proftpd/tls.log
    TLSRSACertificateFile       /etc/pki/tls/proftpd/proftpd.pem
    TLSRSACertificateKeyFile    /etc/pki/tls/proftpd/proftpd.key
	PassivePorts 50000 50010
</IfModule>
<IfModule mod_ifsession.c>
    <IfUser ftpuser>
        TLSRequired                 off
    </IfUser>
</IfModule>
```

**Comment by Castaglia** [comment]
What does [ProFTPD debug logging](http://www.proftpd.org/docs/howto/Debugging.html) show when the user logs in?  What does the `TLSLog` show?

**Comment by viktor077** [comment]
**Debug log**
```
proftpd_1  | 2023-06-01 19:52:13,150 srvr-ftp proftpd[1] 127.0.1.1: ROOT PRIVS at mod_ctrls.c:759
proftpd_1  | 2023-06-01 19:52:13,150 srvr-ftp proftpd[1] 127.0.1.1: RELINQUISH PRIVS at mod_ctrls.c:765
proftpd_1  | 2023-06-01 19:52:16,424 srvr-ftp proftpd[7] 127.0.1.1: ROOT PRIVS at main.c:1301
proftpd_1  | 2023-06-01 19:52:16,424 srvr-ftp proftpd[7] 127.0.1.1: RELINQUISH PRIVS at main.c:1305
proftpd_1  | 2023-06-01 19:52:16,424 srvr-ftp proftpd[7] 127.0.1.1: no matching vhost found for ::ffff:192.168.0.2#21, using 'ftp' listening on wildcard address
proftpd_1  | 2023-06-01 19:52:16,424 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ROOT PRIVS at main.c:1110
proftpd_1  | 2023-06-01 19:52:16,424 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): SETUP PRIVS at main.c:1115
proftpd_1  | 2023-06-01 19:52:16,424 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): session requested from client in unknown class
proftpd_1  | 2023-06-01 19:52:16,424 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): performing module session initializations
proftpd_1  | 2023-06-01 19:52:16,424 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ROOT PRIVS at ../modules/mod_tls.c:6749
proftpd_1  | 2023-06-01 19:52:16,425 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RELINQUISH PRIVS at ../modules/mod_tls.c:6752
proftpd_1  | 2023-06-01 19:52:16,425 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ROOT PRIVS at ../modules/mod_tls.c:12735
proftpd_1  | 2023-06-01 19:52:16,425 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RELINQUISH PRIVS at ../modules/mod_tls.c:12738
proftpd_1  | 2023-06-01 19:52:16,425 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ROOT PRIVS at ../modules/mod_tls.c:3831
proftpd_1  | 2023-06-01 19:52:16,425 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RELINQUISH PRIVS at ../modules/mod_tls.c:3861
proftpd_1  | 2023-06-01 19:52:16,449 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ROOT PRIVS at ../modules/mod_tls.c:17916
proftpd_1  | 2023-06-01 19:52:16,450 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RELINQUISH PRIVS at ../modules/mod_tls.c:18009
proftpd_1  | 2023-06-01 19:52:16,450 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): mod_tls/2.9.2: supporting SSLv3, TLSv1, TLSv1.1, TLSv1.2, TLSv1.3 protocols
proftpd_1  | 2023-06-01 19:52:16,450 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ROOT PRIVS at ../modules/mod_sql.c:5542
proftpd_1  | 2023-06-01 19:52:16,450 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RELINQUISH PRIVS at ../modules/mod_sql.c:5545
proftpd_1  | 2023-06-01 19:52:16,450 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): mod_sql/4.5: defaulting to 'mysql' backend
proftpd_1  | 2023-06-01 19:52:16,451 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ROOT PRIVS at ../modules/mod_quotatab.c:4435
proftpd_1  | 2023-06-01 19:52:16,451 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RELINQUISH PRIVS at ../modules/mod_quotatab.c:4437
proftpd_1  | 2023-06-01 19:52:16,451 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ROOT PRIVS at ../modules/mod_quotatab.c:4453
proftpd_1  | 2023-06-01 19:52:16,451 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RELINQUISH PRIVS at ../modules/mod_quotatab.c:4455
proftpd_1  | 2023-06-01 19:52:16,451 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ROOT PRIVS at mod_delay.c:2107
proftpd_1  | 2023-06-01 19:52:16,451 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RELINQUISH PRIVS at mod_delay.c:2110
proftpd_1  | 2023-06-01 19:52:16,451 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ROOT PRIVS at mod_auth.c:215
proftpd_1  | 2023-06-01 19:52:16,451 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): opening scoreboard '/run/proftpd/proftpd.scoreboard'
proftpd_1  | 2023-06-01 19:52:16,451 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RELINQUISH PRIVS at mod_auth.c:217
proftpd_1  | 2023-06-01 19:52:16,451 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): AuthOrder in effect, resetting auth module order
proftpd_1  | 2023-06-01 19:52:16,451 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): connected - local  : ::ffff:192.168.0.2:21
proftpd_1  | 2023-06-01 19:52:16,451 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): connected - remote : 192.168.0.1:46219
proftpd_1  | 2023-06-01 19:52:16,451 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): FTP session opened.
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): setting CommandBufferSize to 512
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching PRE_CMD command 'USER ftpuser' to mod_tls
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching PRE_CMD command 'USER ftpuser' to mod_core
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching PRE_CMD command 'USER ftpuser' to mod_core
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching PRE_CMD command 'USER ftpuser' to mod_delay
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching PRE_CMD command 'USER ftpuser' to mod_auth
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching CMD command 'USER ftpuser' to mod_ratio
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching CMD command 'USER ftpuser' to mod_auth
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching POST_CMD command 'USER ftpuser' to mod_sql
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching POST_CMD command 'USER ftpuser' to mod_ifsession
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching POST_CMD command 'USER ftpuser' to mod_delay
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching LOG_CMD command 'USER ftpuser' to mod_sql
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching LOG_CMD command 'USER ftpuser' to mod_log
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching LOG_CMD command 'USER ftpuser' to mod_delay
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching PRE_CMD command 'PASS (hidden)' to mod_tls
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching PRE_CMD command 'PASS (hidden)' to mod_core
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching PRE_CMD command 'PASS (hidden)' to mod_core
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching PRE_CMD command 'PASS (hidden)' to mod_sql_passwd
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching PRE_CMD command 'PASS (hidden)' to mod_sql
proftpd_1  | 2023-06-01 19:52:16,452 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching PRE_CMD command 'PASS (hidden)' to mod_delay
proftpd_1  | 2023-06-01 19:52:16,453 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching PRE_CMD command 'PASS (hidden)' to mod_auth
proftpd_1  | 2023-06-01 19:52:16,453 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching CMD command 'PASS (hidden)' to mod_auth
proftpd_1  | 2023-06-01 19:52:16,509 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): retrieved UID 1001 for user 'ftpuser'
proftpd_1  | 2023-06-01 19:52:16,509 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): no supplemental groups found for user 'ftpuser'
proftpd_1  | 2023-06-01 19:52:16,510 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): user 'ftpuser' authenticated by mod_sql.c
proftpd_1  | 2023-06-01 19:52:16,510 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ROOT PRIVS at auth.c:1774
proftpd_1  | 2023-06-01 19:52:16,510 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RELINQUISH PRIVS at auth.c:1777
proftpd_1  | 2023-06-01 19:52:16,510 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): USER PRIVS 1001 at mod_auth.c:1577
proftpd_1  | 2023-06-01 19:52:16,511 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RELINQUISH PRIVS at mod_auth.c:1579
proftpd_1  | 2023-06-01 19:52:16,511 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]):
proftpd_1  | 2023-06-01 19:52:16,511 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): Config for ftp:
proftpd_1  | 2023-06-01 19:52:16,512 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): <IfUser>
proftpd_1  | 2023-06-01 19:52:16,512 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]):  TLSRequired
proftpd_1  | 2023-06-01 19:52:16,513 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): Limit
proftpd_1  | 2023-06-01 19:52:16,513 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]):  DenyAll
proftpd_1  | 2023-06-01 19:52:16,513 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): DefaultServer
proftpd_1  | 2023-06-01 19:52:16,514 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ServerIdent
proftpd_1  | 2023-06-01 19:52:16,514 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): DenyFilter
proftpd_1  | 2023-06-01 19:52:16,514 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): AllowStoreRestart
proftpd_1  | 2023-06-01 19:52:16,515 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): AllowRetrieveRestart
proftpd_1  | 2023-06-01 19:52:16,515 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ScoreboardScrub
proftpd_1  | 2023-06-01 19:52:16,515 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): Port
proftpd_1  | 2023-06-01 19:52:16,515 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): Umask
proftpd_1  | 2023-06-01 19:52:16,516 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): DirUmask
proftpd_1  | 2023-06-01 19:52:16,516 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): CommandBufferSize
proftpd_1  | 2023-06-01 19:52:16,516 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): UserID
proftpd_1  | 2023-06-01 19:52:16,516 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): UserName
proftpd_1  | 2023-06-01 19:52:16,517 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): GroupID
proftpd_1  | 2023-06-01 19:52:16,517 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): GroupName
proftpd_1  | 2023-06-01 19:52:16,517 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): AllowOverwrite
proftpd_1  | 2023-06-01 19:52:16,518 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): AuthOrder
proftpd_1  | 2023-06-01 19:52:16,518 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): Protocols
proftpd_1  | 2023-06-01 19:52:16,518 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RequireValidShell
proftpd_1  | 2023-06-01 19:52:16,518 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): MaxClients
proftpd_1  | 2023-06-01 19:52:16,519 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): MaxClientsPerHost
proftpd_1  | 2023-06-01 19:52:16,519 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): MaxLoginAttempts
proftpd_1  | 2023-06-01 19:52:16,519 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): TimeoutIdle
proftpd_1  | 2023-06-01 19:52:16,519 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): TimeoutLogin
proftpd_1  | 2023-06-01 19:52:16,519 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): TimeoutNoTransfer
proftpd_1  | 2023-06-01 19:52:16,519 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): TimeoutStalled
proftpd_1  | 2023-06-01 19:52:16,519 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): DefaultTransferMode
proftpd_1  | 2023-06-01 19:52:16,519 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): AllowForeignAddress
proftpd_1  | 2023-06-01 19:52:16,519 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RootLogin
proftpd_1  | 2023-06-01 19:52:16,519 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): AccessDenyMsg
proftpd_1  | 2023-06-01 19:52:16,520 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): AccessGrantMsg
proftpd_1  | 2023-06-01 19:52:16,520 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): WtmpLog
proftpd_1  | 2023-06-01 19:52:16,520 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): SQLAuthTypes
proftpd_1  | 2023-06-01 19:52:16,520 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): SQLPasswordEngine
proftpd_1  | 2023-06-01 19:52:16,520 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): SQLAuthenticate
proftpd_1  | 2023-06-01 19:52:16,520 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): SQLConnectInfo
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): SQLUserTable
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): SQLUsernameField
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): SQLPasswordField
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): SQLUidField
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): SQLGidField
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): SQLHomedirField
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): SQLShellField
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): SQLLogFile
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RequireValidShell
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): QuotaEngine
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): QuotaDirectoryTally
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): QuotaDisplayUnits
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): QuotaShowQuotas
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): TLSEngine
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): TLSOptions
proftpd_1  | 2023-06-01 19:52:16,521 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): TLSRequired
proftpd_1  | 2023-06-01 19:52:16,523 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): TLSVerifyClient
proftpd_1  | 2023-06-01 19:52:16,523 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): TLSProtocol
proftpd_1  | 2023-06-01 19:52:16,523 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): TLSOptions
proftpd_1  | 2023-06-01 19:52:16,523 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): TLSLog
proftpd_1  | 2023-06-01 19:52:16,523 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): TLSRSACertificateFile
proftpd_1  | 2023-06-01 19:52:16,523 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): TLSRSACertificateKeyFile
proftpd_1  | 2023-06-01 19:52:16,523 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): PassivePorts
proftpd_1  | 2023-06-01 19:52:16,523 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ROOT PRIVS at mod_auth.c:1673
proftpd_1  | 2023-06-01 19:52:16,523 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): opening TransferLog '/var/log/xferlog'
proftpd_1  | 2023-06-01 19:52:16,523 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): RELINQUISH PRIVS at mod_auth.c:1718
proftpd_1  | 2023-06-01 19:52:16,524 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): ROOT PRIVS at mod_auth.c:1769
proftpd_1  | 2023-06-01 19:52:16,524 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): SETUP PRIVS at mod_auth.c:1791
proftpd_1  | 2023-06-01 19:52:16,524 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): in dir_check_full(): path = '/ftp/ftpuser', fullpath = '/ftp/ftpuser'
proftpd_1  | 2023-06-01 19:52:16,524 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching POST_CMD command 'PASS (hidden)' to mod_sql
proftpd_1  | 2023-06-01 19:52:16,524 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching POST_CMD command 'PASS (hidden)' to mod_ifsession
proftpd_1  | 2023-06-01 19:52:16,524 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): mod_ifsession/1.3.1: merging <IfUser ftpuser> directives in
proftpd_1  | 2023-06-01 19:52:16,524 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): in dir_check_full(): path = '/ftp/ftpuser', fullpath = '/ftp/ftpuser'
proftpd_1  | 2023-06-01 19:52:16,525 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): dispatching POST_CMD command 'PASS (hidden)' to mod_tls
proftpd_1  | 2023-06-01 19:52:16,528 srvr-ftp proftpd[7] 127.0.1.1 (192.168.0.1[192.168.0.1]): FTP session closed.

```
**TLS log shows only one row**
```
2023-06-01 19:52:16,525 mod_tls/2.9.2[7]: ftps protocol denied by Protocols config

```

**Comment by Castaglia** [comment]
Thanks!  I'll see why that "ftps protocol denied by Protocols config" message is being logged; I wouldn't expect that.  The rest of the debug logging looks correct.

Given that, what happens when your user/client attempts to do a data transfer?  That is, the above debug logging shows user `ftpuser` authenticating without TLS, which is what your `<IfUser>` configuration allows, so it seems like it is working as expected.

**Comment by viktor077** [comment]
Thanks.
Filezilla client show me this log. 
```
Status:	Connecting to 192.168.0.2:21...
Status:	Connection established, waiting for welcome message...
Response:	220 ftp
Command:	USER ftpuser
Response:	331 Password required for ftpuser
Command:	PASS ******
Response:	530 Login incorrect.
Error:	Critical error: Could not connect to server
```

**Comment by Castaglia** [comment]
Got it.  This may be a bug in the handling of the `Protocols` directive in the `mod_tls` module; I'll try to reproduce this later today.  In the mean time, to get you unblocked, you might comment out the `Protocols` directive in your `proftpd.conf` for now.

**Comment by Castaglia** [comment]
This should now be fixed in the master branch; the fix has been backported to the 1.3.8 branch.  Thanks!
